app/models/post.rb

```diff
@@ -1,3 +1,9 @@
 class Post < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User', counter_cache: :posts_counter
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
 end
```

Comment:
[Suggestion] This is a suggestion that should be taken into serious consideration . It would be nice to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_posts_counter method every time a new post has been saved to database, then make the update_posts_counter method private.